### ISSUE TITLE
[bug] console.log(new Date(event.toDateString()));  // not working

### ISSUE THREAD

**Issue by Guang1234567** [facebook/hermes#794]
Hello dev:

Below code is not working in react-native 0.68  when `enableHermes: true` on android platform.

```js
console.log(new Date(event.toDateString()));  // `Invalid Date` in react-native 0.68
```

![image](https://user-images.githubusercontent.com/8240718/184813582-2b434b6b-16e2-4d72-bae4-14316daecfeb.png)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toDateString

**Comment by jpporto** [comment]
Hi @Guang1234567,

When I run

```
const event = new Date(1993, 6, 28, 14, 39, 7);

console = { log: print };

console.log(event.toString());
// expected output: Wed Jul 28 1993 14:39:07 GMT+0200 (CEST)
// (note: your timezone may vary)

console.log(event.toDateString());
// expected output: Wed Jul 28 1993

console.log(new Date(event.toDateString()));  // `Invalid Date` in react-native 0.68
```

in the hermes cli I get

```
Wed Jul 28 1993 14:39:07 GMT-0700
Wed Jul 28 1993
Wed Jul 28 1993 00:00:00 GMT-0700
```

Which is consistent with the output I get in MDN. What issue are you seeing?

John Paul

**Comment by Guang1234567** [comment]
Hi @jpporto 

My fault, not `react-native 0.68`.

My company's project(can not upgrade to recent version of RN) is base on `"react-native": "0.63.4"`.  The  issue be reproduced  on  `"react-native": "0.63.4"`.

Thanks your reply.

**Comment by jpporto** [comment]
This is really a reactnative issue as you're consuming the hermes that shipped with release 0.63. I am afraid there's not much to be done here, other than upgrading to the most recent RN.

What are the blockers preventing your company to switch to a more recent RN version?

**Comment by Guang1234567** [comment]
A temporary solution for RN 0.63.4.

```diff
- enableHermes: true
+ enableHermes: false
```

> What are the blockers preventing your company to switch to a more recent RN version?

Is shame to soft engine has no rights of speaking in xxx,  O(∩_∩)O.

Thanks your reply.

**Comment by neildhar** [comment]
This is the same issue as https://github.com/facebook/hermes/issues/578. The first RN release with the fix is 0.66.

Note that parsing this date format is non-standard, ISO 8601 date strings are the best way to get consistent behaviour across JS engines.
